File: math/integer-to-roman/integer-to-roman.ts
function intToRoman(num: number): string {
  const values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
  const roman = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I'];

  const sb: string[] = [];

  for (let i = 0; i < values.length; i++) {
    const value: number = values[i];

    let quotient: number = Math.floor(num / value);

    if (quotient > 0) {
      sb.push(roman[i].repeat(quotient));
      num %= quotient * value;
    }

    if (num === 0) {
      return sb.join('');
    }
  }
  return sb.join('');
}

File: math/integer-to-roman/integer-to-roman.go
package integertoroman

import "strings"

func intToRoman(num int) string {
	values := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
	roman := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}

	sb := []string{}

	for i, value := range values {
		quotient := num / value
		if quotient > 0 {
			for j := 0; j < quotient; j++ {
				sb = append(sb, roman[i])
			}
			num %= (quotient * value)
		}

		if num == 0 {
			return strings.Join(sb, "")
		}
	}
	return strings.Join(sb, "")
}
